code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+
+class FreshdeskPage extends Page
+{
+}
+
+class FreshdeskPage_Controller extends Page_Controller
+{
+    /**
+     * Status mapping for Freshdesk tickets.
+     *
+     * @var array
+     * @config
+     */
+    private static $freshdeskStatus = [
+        2 => 'open',
+        3 => 'pending',
+        4 => 'resolved',
+        5 => 'closed',
+    ];
+
+    /**
+     * Priority mapping for Freshdesk tickets.
+     *
+     * @var array
+     * @config
+     */
+    private static $freshdeskPriority = [
+        1 => 'low',
+        2 => 'medium',
+        3 => 'high',
+        4 => 'urgent',
+    ];
+
+    /**
+     * @var array
+     */
+    private static $_tickets = [];
+
+    /*
+    * Returns Freshdesk tickets for a user can be filtered by priority or status
+    *
+    * @param Array $filter
+    * @return PaginatedList
+    *
+    */
+    public function getFreshdeskTickets()
+    {
+        $currentMember = \Member::currentUser();
+        if (!$currentMember || !$currentMember->exists()) {
+            return \Security::permissionFailure();
+        }
+
+        $tickets = $this->getTickets($currentMember);
+
+        if (empty($tickets) || 0 == count($tickets)) {
```

Comment:
Please don't use yoda style. It reads badly.
Also, checking for empty will also return true if an array is empty, so the count check is redundant.

Reply:
This is what the php fixer changed it to, I'll see if there's a setting in there I can change

Reply:
ack